waku/v2/node/rest/relay/api_types.nim

```diff
@@ -0,0 +1,121 @@
+{.push raises: [ Defect ].}
+
+import
+  std/[sets, strformat],
+  stew/byteutils,
+  chronicles,
+  json_serialization,
+  json_serialization/std/options,
+  presto/[route, client, common]
+import ".."/serdes
+import ../../wakunode2
+
+
+#### Types
+
+type
+    PubSubTopicString* = distinct string
+    ContentTopicString* = distinct string
+
+type RelayWakuMessage* = object
```

Comment:
> Also, the naming might be confusing. If I understand correctly, It is basically an alternative interpretation of a WakuMessage.
But, `RelayWakuMessage` might suggest it is related to the message protobuf in the relay spec.

The idea here was to "scope" the WakuMessage REST API model for the Relay API. I did not want it to clash with the same model used in the filer or the store REST APIs. I am still thinking about which is the best strategy to avoid clashes. As we move forward, I think we'll be able to improve the implementation details like this one 😁

Reply:
Thanks for clarifying! Makes sense to me :+1: 
Still, I think the naming is confusing. It might help if the name containted `rest` to make clear that it is associated with the REST api model.